tests/processor/test_loose_artifact_analyzer.py

```diff
@@ -0,0 +1,156 @@
+        test_dir = os.path.join(layer_dir, 'test_dir')
+        os.makedirs(test_dir)
+
+        test_file = os.path.join(test_dir, 'test_file.txt')
+        open(test_file, 'a').close()
+
+        rpm_test_file = os.path.join(test_dir, 'test.rpm')
+        open(rpm_test_file, 'a').close()
+
+        analyzer = LooseArtifactAnalyzer(temp_dir)
+        analyzer._koji_session = m_koji
+        analyzer.run()
+
+        assert os.path.exists(test_file) is True
+        assert os.path.exists(rpm_test_file) is False
+
+    assert m_read_metadata_file.call_count == 1
+    assert m_koji.getRPM.call_count == 1
+
+    container_artifact = content.Artifact.nodes.get(type_='container')  # there should be only 1
+    loose_rpm_artifact = content.Artifact.nodes.get(archive_id=RPM_INFO['id'])
+
+    assert container_artifact.embedded_artifacts.is_connected(loose_rpm_artifact)
+
+    # assert the build was created and connected
+    build = content.Build.nodes.get(id_=str(RPM_INFO['build_id']))
+    assert build.id_ == str(RPM_INFO['build_id'])
+    assert build.type_ == 'build'
+    assert loose_rpm_artifact.build.is_connected(build)
+
+
+@mock.patch('assayist.processor.base.Analyzer.read_metadata_file')
+def test_archives_on_container_layer(m_read_metadata_file):
+    """Test the LooseArtifactAnalyzer.run() function on an unpacked container layer."""
+    build = BuildFactory.create()
+    container = ArtifactFactory.create(type_='container')
+    build.artifacts.connect(container)
+
+    m_read_metadata_file.return_value = {'id': 1}
+
+    m_koji = mock.Mock()
+    m_koji.multiCall.return_value = [[[ARCHIVE_INFO]]]
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        layer_dir = os.path.join(
+            temp_dir, 'unpacked_archives', 'container_layer', container.filename)
+        test_dir = os.path.join(layer_dir, 'test_dir')
+        os.makedirs(test_dir)
+
+        def create_test_file(extension):
+            x = os.path.join(test_dir, 'test_file.' + extension)
+            open(x, 'a').close()
+            return x
+
+        test_file = create_test_file('txt')
+        jar_test_file = create_test_file('jar')
+        tar_test_file = create_test_file('tar')
+        xml_test_file = create_test_file('pom.xml')
+        pom_test_file = create_test_file('pom')
+
+        analyzer = LooseArtifactAnalyzer(temp_dir)
+        analyzer._koji_session = m_koji
+        analyzer.run()
+
+        assert os.path.exists(test_file) is True
+        assert os.path.exists(jar_test_file) is False
+        assert os.path.exists(tar_test_file) is False
+        assert os.path.exists(xml_test_file) is False
+        assert os.path.exists(pom_test_file) is False
+
+    assert m_read_metadata_file.call_count == 1
+    assert m_koji.listArchives.call_count == 4
+
+    container_artifact = content.Artifact.nodes.get(type_='container')  # there should be only 1
+    # We connected four "files" but since the mock method returned the same archive_info for
```

Comment:
Optional: Using `side_effect` on the mock could alleviate this but it is more test data you have to create/modify.

Reply:
Yeah, I think it's fine as-is though.